kivy/uix/behaviors/hover.py

```diff
@@ -0,0 +1,284 @@
+                pos: self.pos
+
+    HovSlider:
+        id: hovslider
+        size_hint: 0.9, 0.1
+        pos_hint: {'center': (0.5, 0.7)}
+        value_track: True
+        value_track_color: (0.95, 0.1, 0.0, 1) if self.hovering or self.grabbed_hover else (0.3, 0.3, 0.5, 1)
+        min: 0
+        max: 1000
+        value: int(self.max * 0.5)
+        hov_callback: lambda value: setattr(root, "hov_grab", value)
+
+    Label:
+        size_hint_y: None
+        pos: 0, hovslider.top + self.height
+        text: "Slider that grabs hover and prevents other widgets from displaying hover effects when being slided"
+
+    BoxLayout:
+        id: sizebox
+        size_hint: 1, 0.12
+        pos: 0, tggrab.top
+        orientation: 'horizontal'
+
+        HovButton:
+            size_hint_x: 0.5
+            hover_height: 10
+            text: 'Set min_hover_height: 10'
+            on_release: root.min_hov = self.hover_height
+
+        HovButton:
+            size_hint_x: 0.5
+            hover_height: 0
+            text: 'Set min_hover_height: 0'
+            on_release: root.min_hov = self.hover_height
+
+    Label:
+        size_hint: 0.9, None
+        height: self.texture_size[1]
+        pos: root.width * 0.05, sizebox.top + (self.height * 0.5)
+        text_size: self.size
+        text: "Buttons that change HoverBehavior min_hover_height to their own height, useful when you want to disable hovering for lower widgets after opening a popup or putting something else on top"
+
+    HovButton:
+        id: tggrab
+        size_hint: 1, 0.12
+        pos: 0, 0
+        text: 'Toggle grab hover'
+        on_release: setattr(root, "hov_grab", None) if root.hov_grab else setattr(root, "hov_grab", self)
+""")
+
+runTouchApp(RootWidget())
+
+
+See :class:`~kivy.uix.behaviors.HoverBehavior` for details.
+'''
+
+__all__ = ('HoverBehavior', )
+
+from kivy.properties import BooleanProperty, NumericProperty, ObjectProperty
+from kivy.weakproxy import WeakProxy
+from kivy.core.window import Window
+
+
+class HoverBehavior(object):
+    '''
+    The HoverBehavior `mixin <https://en.wikipedia.org/wiki/Mixin>`_ provides
+    Hover behavior. When combined with a widget, hovering mouse cursor
+    above it's position will call it's on_hovering event. Please see
+    the :mod:`hover behaviors module <kivy.uix.behaviors.hover>` documentation
+    for more information.
+
+    .. versionadded:: 1.10.1
+    '''
+
+    _hover_grab_widget = None
+    '''WeakProxy of widget which has grabbed focus or None'''
+
+    _hover_widgets = []
+    '''List of hover widget WeakProxy references'''
+
+    min_hover_height = 0
+    '''Numeric class attribute of minimum height where "hovering" property
+    will be updated'''
+
+    hovering = BooleanProperty(False)
+    '''Hover state, is True when mouse enters it's position
+
+    :attr:`hovering` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to `False`.
+    '''
+
+    hover_height = NumericProperty(0)
```

Comment:
This seems to me like implementing a `Z` in Kivy, where all position is 2D which probably isn't the best thing to do, but anyway... maybe it'd be a better idea to put the refs/widgets/whatever into a dictionary (quicker lookup) and use list for storing the *order* of the widgets instead of "height". Then you could use the order with a behavior similar to `Widget.children` and just move the widgets' order if necessary. Maybe even better would be to cache the widgets' position to that order list and update it with widgets' `on_pos`. It'll still be slow(or better said executed too many times), but you wouldn't need to touch the widgets alone, just the list. Then again if you use it with `Animation` it'll go to hell anyway.